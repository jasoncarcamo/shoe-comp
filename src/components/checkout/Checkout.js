import React from "react";
import CartContext from "../CartContext/CartContext";
import "./checkout.css";
import Shoe from "../shoe/Shoe";




export default class Checkout extends React.Component{
    constructor(props){
        super(props);
        this.state = {
            items: []
        };

    }

    
    static contextType = CartContext;

    componentDidMount(){

    }

    renderEditSelection = ()=>{

        const orders = this.context.items;
        
        if(orders.length === 0){
            return <span id="no-items">No items in your cart.</span>;
        };

        return orders.map( (order, index)=> {

            const removeItem = ()=>{
                this.context.removeItem(index)
            };

            return (
            <li key={index} className="checkout-item">

                <h3>Order: {index + 1}</h3>

                <div className="checkout-items">
                    <div className="checkout-features">
                        <button className="checkout-remove-item" type="button" onClick={removeItem}><span>-</span></button>

                        <Shoe className="checkout-SVG" top={order.top} middle={order.middle} bottom={order.bottom}></Shoe>

                        <button type="button" onClick={()=>{ this.props.history.push(`/editshoe?item=${index + 1}`)}}>Edit</button>
                    </div>
                    
                    <div className="checkout-item-info">
                        <p>Size: {order.size}</p>
                        <p>Quatity: {order.quantity}</p>
                    </div>
                </div>

            </li>)});
    }

    renderOrderButtons = ()=>{
        return (
            <>
                <button id="checkout-place-order" type="buton" onClick={this.handlePlaceOrder}>Place order</button>
                <button id="checkout-cancel-order" type="button" onClick={this.handleCancelOrder}>Cancel order</button>
            </>
        )
    }

    handleCancelOrder = ()=>{
        this.context.clearCheckout();
    }

    handlePlaceOrder = ()=>{
        this.context.placeOrder();
        this.props.history.push('/user');
    }

    render(){
        
        return (
            <section id="checkout-section">
                <ul>
                    {this.renderEditSelection()}
                    {this.context.items.length !== 0 ? this.renderOrderButtons() : ""}
                </ul>
            </section>
        )
    }
}