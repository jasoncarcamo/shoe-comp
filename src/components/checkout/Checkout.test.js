import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import Checkout from "./Checkout";
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter()});



describe("<Checkout/>", ()=>{
    
    it("Renders without crashing", ()=>{
        const div = document.createElement("div");

        ReactDOM.render(<BrowserRouter><Checkout/></BrowserRouter>, div);

        ReactDOM.unmountComponentAtNode(div);
    });

    it("Creates a snapshot", ()=>{
        const tree = renderer.create(<BrowserRouter><Checkout/></BrowserRouter>).toJSON();

        expect(tree).toMatchSnapshot();
    });

})