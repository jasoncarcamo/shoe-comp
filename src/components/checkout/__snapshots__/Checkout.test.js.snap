// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkout/> Creates a snapshot 1`] = `
<section
  id="checkout-section"
>
  <ul>
    <span
      id="no-items"
    >
      No items in your cart.
    </span>
    
  </ul>
</section>
`;
